nv_peer_mem.c

```diff
@@ -138,12 +141,26 @@ static void nv_get_p2p_free_callback(void *data)
 	    in case it's called internally by that callback.
 	*/
 	page_table = nv_mem_context->page_table;
+
+#if NV_DMA_MAPPING
+	if (!nv_mem_context->dma_mapping) {
+		peer_err("nv_get_p2p_free_callback -- invalid dma_mapping\n");
+		goto out;
+	}
+	dma_mapping = nv_mem_context->dma_mapping;
+#endif
+
 	/* For now don't set nv_mem_context->page_table to NULL, 
 	  * confirmed by NVIDIA that inflight put_pages with valid pointer will fail gracefully.
 	*/
 
 	ACCESS_ONCE(nv_mem_context->is_callback) = 1;
 	(*mem_invalidate_callback) (reg_handle, nv_mem_context->core_context);
+#if NV_DMA_MAPPING
+	ret = nvidia_p2p_free_dma_mapping(dma_mapping); 
```

Comment:
it works as nvidia_p2p_put_pages, dma_unmapping is implicitly done by NV driver after invalidation callback, but client code is in charge of freeing the dma_mapping obj.
so no, we don't need to call dma_unmap here.